src/main.rs

```diff
@@ -13,44 +15,49 @@ fn get_far_from_cel(cel: &f64) -> f64 {
     cel * (9.0 / 5.0) + 32.0
 }
 
-fn main() {
-    let far_symbol = String::from("f");
-    let cel_symbol = String::from("c");
+fn get_value_from_user() -> Result<f64, num::ParseFloatError> {
+    let mut value = String::new();
+    io::stdin()
+        .read_line(&mut value)
+        .expect("An error when typing value");
+    let numeric_value: f64 = value.trim().parse()?;
+    Ok(numeric_value)
+}
 
-    let source: Mode;
+struct Symbols {
+    far: String,
+    cel: String,
+}
 
-    println!(
-        "You want to put celcius of fahrenheits? {}/{}",
-        cel_symbol, far_symbol
-    );
+fn get_temp_sign_from_user(symbols: Symbols) -> Mode {
+    let mut source = Mode::None;
 
-    loop {
+    while matches!(source, Mode::None) {
         let mut mode = String::new();
-        io::stdin().read_line(&mut mode).expect("An error while reading temperature sign");
+        io::stdin()
+            .read_line(&mut mode)
+            .expect("An error while reading temperature sign");
 
         mode = String::from(mode.trim());
 
-        if mode == far_symbol {
+        if mode == symbols.far {
             source = Mode::Far;
             break;
         }
 
-        if mode == cel_symbol {
+        if mode == symbols.cel {
             source = Mode::Celc;
             break;
         }
 
-        println!("Please, press only {} or {}", far_symbol, cel_symbol)
+        println!("Please, press only {} or {}", symbols.far, symbols.cel)
     }
```

Comment:
this could become something like:
```rust
match mode {
far_symbol => {
Mode::Far
}
cel_symbol => {
Mode::Celc
}
_ => println!("Please try again")
}
```

Reply:
Seems like there is a restriction in Rust of matching against the result of variables comparison, that's the best I can get

https://github.com/Komalov/temperature_convertor/pull/1/commits/e3c6638bc4fe62c0de983b5d1b4cd3ee286c6943

Reply:
That's okay, there's a better way of doing it but we'll get there!